libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -110,10 +114,10 @@ export class GRPCService implements Service {
     try {
       const { response } = await this.client.resolveObject({
         flagKey,
-        context: Struct.fromJsonString(JSON.stringify(context)),
+        context: this.convertContext(context),
       });
       return {
-        value: response.value as T,
+        value: (response.value ? Struct.toJson(response.value) : undefined) as T,
```

Comment:
This is the actual fix.

Reply:
Should we throw if value is undefined? I think an undefined value is invalid.

Reply:
I wasn't sure about this. Why do you think an undefined value is invalid?

Reply:
The value is required but since it's a generic I suppose technically it can be undefined.

https://github.com/open-feature/node-sdk/blob/main/src/types.ts#L202

It just doesn't seem like the behavior I would expect. In what situation would an undefined value be useful? Is this something we had to worry about with the Java provider?

Reply:
> The value is required but since it's a generic I suppose technically it can be undefined.

That's part of the problem. I think we need to refine the type to not be generic, but something more representative of structured data, like it is in the Java/dotnet SDK. In Java/dotnet, a node (`Value`) can be returned which represents a JSON `null`. I think we should refine the type to be more like the `Evaluation Context` type, which extends this:

`export type JSONValue = null | string | number | boolean | Date | { [x: string]: JSONValue } | Array<JSONValue>;`.

This is also more in-line with the other SDKs, where the context and the object resolver return type are similarly structured (though not exactly the same).

> It just doesn't seem like the behavior I would expect. In what situation would an undefined value be useful? Is this something we had to worry about with the Java provider?

In other SDKs we've structured things very flexibly, so anything like a JSON node can be returned (this also worked great for other data-interchange formats, see [GRPC's object type](https://github.com/timostamm/protobuf-ts/blob/3eea598471fb9c891c8b6da06bc1cbe18bc3457e/packages/runtime/src/json-typings.ts#L10)) so `null`ish values make sense. To your point, this isn't the same as undefined though. I'm not sure what would happen if the value returned in the GRPC message was completely missing in Java.

Reply:
Throwing parse error now.